Assets/Scripts/Enemy/EnemyService.cs

```diff
@@ -43,8 +46,12 @@ public void SpawnEnemyTank()
             enemySpawnPos = enemySpawnPos.SetRandomVectorXYZ(-39, 39, 0, 10, -39, 39);
             enemySpawnPos = enemySpawnPos.SetY(-3.68f);
             EnemyTankPositions.Add(enemySpawnPos);
-            GameObject enemyTank = Instantiate(EnemyTankViews[Random.Range(0, EnemyTankViews.Count)], enemySpawnPos, Quaternion.identity) as GameObject;
-            InstantiatedEnemyTanks.Add(enemyTank);
+            GameObject enemyTank = enemyPoolService.GetEnemyTank(EnemyTankViews[Random.Range(0, EnemyTankViews.Count)]);
```

Comment:
same thing as above - pool should be of a specific class (controller or view) not gameobject

Reply:
Done Created A Pool of my EnemyBehaviour script ie. the view script for enemy